src/Rating.js

```diff
@@ -1,8 +1,191 @@
 'use strict';
 
-import RatingBase from './Rating.soy';
+import core from 'metal';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+import templates from './Rating.soy';
 
-class Rating extends RatingBase {
+class Rating extends Component {
+    /**
+     * @inheritDoc
+     */
+    constructor(opt_config) {
+        super(opt_config);
+
+        this.ratingClicked_ = this.value;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    attached() {
+        this.delegate('mouseover', '.rating-item', this.handleMouseOverEvent.bind(this));
+        this.delegate('click', '.rating-item', this.handleClickEvent.bind(this));
+        this.on('mouseleave', this.handleMouseLeaveEvent.bind(this));
+    }
+
+    /**
+     * Handles click event
+     * @param {Event} event
+     * @protected
+     */
+    handleClickEvent(event) {
+        if (!this.disabled) {
+            let index = parseInt(event.delegateTarget.getAttribute('data-index'), 10);
+
+            if (index === this.ratingClicked_ && this.canReset) {
+                this.reset();
+            }
+            else {
+                this.value = index;
+            }
+
+            this.ratingClicked_ = this.value;
+        }
+    }
+
+    /**
+     * Handles mouseleave event
+     * @protected
+     */
+    handleMouseLeaveEvent() {
+        this.currentMouseTarget_ = undefined;
+        this.setPreviousRate_();
+    }
+
+    /**
+     * Handles mouseover event
+     * @param {event} event
+     * @protected
+     */
+    handleMouseOverEvent(event) {
+        if (!this.disabled) {
+            let index = Number.parseInt(event.delegateTarget.getAttribute('data-index'), 10);
+
+            if (this.currentMouseTarget_ !== index) {
```

Comment:
This is one of the branches which is not being covered, since all tests always enter the if clause. From looking at this it seems like it could just be removed instead though. When an attribute is set to the same value it already has nothing happens today, so you shouldn't need to check for that, unless I'm missing some use case (in which case it's good to add a test for this).